### ISSUE TITLE
Need a way to subscribe to network & account changes

### ISSUE THREAD

**Issue by chadoh** [stellar/freighter#846]
### What problem does your feature solve?

Right now, in order to have an app respond to a user's network & account changes in Freighter, an app needs to poll Freighter every X seconds. Example from the dependency used by Soroban's official dapp example: https://github.com/esteblock/soroban-react/blob/73f94af55d26f8ca42533af6c779998caeb3bb19/packages/core/src/SorobanReactProvider.tsx#L84-L127

### What would you like to see?

A way to subscribe to network and account changes. Maybe it could be used like:

```tsx
import { useEffect, useState } from 'react'
import { subscribeToChanges, getPublicKey, getNetwork } from '@stellar/freighter-api'

export function useFreighter() {
  const [publicKey, setPublicKey] = useState(null)
  const [network, setNetwork] = useState(null)

  useEffect(function initialLoad() {
    getPublicKey.then(setPublicKey)
    getNetwork.then(setNetwork)
  }, [])

  useEffect(function subscription() {
    return subscribeToChanges(({ publicKey: newPublicKey, network: newNetwork }) => {
      setPublicKey(newPublicKey)
      setNetwork(newNetwork)
    })
  }, [])

  return { publicKey, network }
}
```

Here we see that `subscribeToChanges` should:

- return a function which allows unsubscribing
- accept a function as an argument which
   - takes an object as an argument that has at least `publicKey` and `network` arguments, maybe also a `networkDetails` (or, better, each key of `networkDetails` could be a separate key in the top level object)
   - returns void

### What alternatives are there?

The heavy-handed polling workarounds that app authors need to use today

**Comment by chadoh** [comment]
We should make sure this functionality works well with [useSyncExternalStore](https://react.dev/reference/react/useSyncExternalStore)

**Comment by GHesericsu** [comment]
+1 on this

**Comment by piyalbasu** [comment]
Implemented in [v5.23.0](https://github.com/stellar/freighter/releases/tag/5.23.0)
